### ISSUE TITLE
Add gltfUpAxis support

### ISSUE THREAD

**Issue by javagl** [CesiumGS/cesium-unreal#315]
There have been several issues reported in the Unreal Marketplace and the Forum about a wrong orientation of models (e.g. https://community.cesium.com/t/unreal-load-3dtileset-not-by-ion-looks-wrong-rotation/12835 ).

This was caused by 3D Tiles assets that relied on the `gltfUpAxis` property in the `asset`:

    "asset": {
      "gltfUpAxis":"Z",
       ...
    }

We will encourage users to ask the creators of these assets to **not** use this property. The glTF specification defines [the +Y-axis as the up-axis](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#coordinate-system-and-units), and the `gltfUpAxis` property is not part of the 3D Tiles specification.

---

Side note: It is usually possible to upgrade such tilesets, using the `upgrade` functionality of [the `3d-tiles-validator` tools in the `2.0-tools` branch](https://github.com/CesiumGS/3d-tiles-validator/tree/2.0-tools/tools). But depending on other aspects of the input data, this can cause other problems, and may not be a viable path for all cases.

---

**Comment by javagl** [comment]
I have created two **DRAFT** PRs with a possible solution:

- https://github.com/CesiumGS/cesium-native/pull/200 in `cesium-native` parses the `gltfUpAxis` property, stores it in the `Tileset`, and passes it into the glTF model
- https://github.com/CesiumGS/cesium-unreal/pull/316 in `cesium-unreal` performs the rotation of the glTF model, depending on the `gltfUpAxis` property

An issue that still has to be decided on is how this information is passed from `cesium-native` to `cesium-unreal`. Right now, the only way to transfer "arbitrary" data to a consumer is via the `extras` field of the glTF model. This was also done for the `RTC_CENTER`, but has some caveats and drawbacks - see https://github.com/CesiumGS/cesium-unreal/issues/310 for details. 

A longer-term solution should allow to include such information without relying on the `extras` of the glTF model. Generally speaking, the function 
```
class CESIUM3DTILES_API IPrepareRendererResources {
  virtual void* prepareInLoadThread(
      const CesiumGltf::Model& model,
      const glm::dmat4& transform) = 0;
```
may need more than just the `CesiumGltf::Model` in the future. This also refers to information about the feature/batch-tables from 3D Tiles that may have to be passed along.

It would be preferable to **not** use the `extras` in the way that they are currently used. If a solution for that can be found before the next release, it could also be used to pass along the information about the `gltfUpAxis`.

**Comment by baothientran** [comment]
it's reasonable to pass tile itself to the `IPrepareRendererResources::prepareInLoadThread()`, other methods already did it, so I think it will be okay in this case

**Comment by kring** [comment]
@baothientran it's unsafe to access a `Tile` from a worker thread. Accessing it from the main/game thread is fine. Have you seen somewhere else where Tile is accessed from a worker?

**Comment by baothientran** [comment]
Ah I just [commented about it](https://github.com/CesiumGS/cesium-native/pull/200#discussion_r608944885) on the other PR and didn't really account for it when writing the comment above. Silly me!

For your question, I haven't seen the tile is captured from a worker actually

**Comment by javagl** [comment]
The corresponding PRs have been updated based on the initial feedback, the build passed, and they are now marked as Ready for Review. If somebody wants to give it a try, that would be great.

(An aside: I did not find an example data set where the `gltfUpAxis` was **not** `"Z"`. But I carefully checked the transform matrices, and *hope* that they are right. These coordinate transforms can be real brain-teasers. Some of the intricacies are e.g. mentioned in https://github.com/CesiumGS/3d-tiles/issues/280 , which I found quite interesting)
